ang/contactlayout/contactlayout.js

```diff
@@ -507,4 +552,48 @@
     };
   });
 
+  // Service for loading relationship type options and displaying loading state.
+  angular.module('contactlayout')
+    .service('contactLayoutRelationshipOptions', function (crmApi) {
+      var relationshipOptionsPromise;
+      var service = this;
+
+      service.isLoading = false;
+      service.options = [];
+
+      // loads and stores the relationship type options.
+      service.loadOptions = function () {
+        service.isLoading = true;
+
+        if (!relationshipOptionsPromise) {
+          relationshipOptionsPromise = crmApi('RelationshipType', 'get', {
+            is_active: 1,
+            sequential: 1,
+            options: { limit: 0 }
+          });
```

Comment:
Please use crmApi4 here.
```suggestion
          relationshipOptionsPromise = crmApi4('RelationshipType', 'get', {
            where: [["is_active", "=", true]]
          });
```